File: htmlportion/oops.js/inheritance.js
//inheritances =class parenti
//extents= class call chey
//multiple inheritance dose'nt support



// single inheritance
//class parent{
  //  phone(){
    //    console.log("have a nokiac1");
    //}
//}
//class child extends parent{

//}
//var ch=new child()
//ch.phone()


//multilevel

class Parent{
     m1(){
        console.log("nokiac1");
    }
}
class Child extends Parent{
        m2(){
            console.log("m2 method");
        }
}
class Subchild extends Child{
    m3(){
        console.log("m3 method");
    }
}
var ch=new Subchild()
ch.m3()
ch.m2()
ch.m1()

File: objectorientprgm/inheritance.py
#single inheritance
class parent:
    def phone(self):
        print("inside parents phone method")
class child(parent):
    pass
ch=child()
ch.phone()
#multilevel inheritance
class parent:
    def m1(self):
        print("inside m1")
class subchild(parent):
    def m2(self):
        print("inside m2")
class child(subchild):
    def m3(self):
        print("inside m3")
ch=child()
ch.m1()
ch.m2()
ch.m3()

#mulitple inheritance
class parent:
    def m1(self):
        print("m1")
class subchild:
    def m2(self):
        print("m2")
class child(subchild,parent):
    def m3(self):
        print("m3")
ch=child()
ch.m1()
ch.m2()
ch.m3()